### PR TITLE
Avoid hardcoding global address spaces

### PR DESCRIPTION
**Pull request by sumesh13**
Doesnt look like LLVM provides any way to map default address spaces. So provide a target API to do that

### CODE REVIEW

**Comment by whchung** [comment]
What about the logic for NVPTX?

**Comment by sumesh13** [comment]
In the upstream, the code is using default address space - so I didnt add any NVPTX code.. They likely need the same mapping but I didnt want to make any assumptions.

**Comment by whchung** [comment]
got it.

File: tensorflow/compiler/xla/service/gpu/target_util.h [hunk 1]
```diff
@@ -56,6 +56,9 @@ enum class TargetFunctionID {
   kFmod,
   kRound,
 };
+// AMDGCN target address spaces
+constexpr int kAMDGPUGlobalMemoryAddrSpace = 1;
+constexpr int kAMDGPUSharedMemoryAddrSpace = 3;
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/target_util.h]: @sumesh13 do you want to amend the PR so it covers `kNVPTXSharedMemoryAddrSpace` and `kAMDGPUSharedMemoryAddrSpace` as well?

Comment (sumesh13) [reply tensorflow/compiler/xla/service/gpu/target_util.h]: Thanks for pointing that out! I incorrectly assumed shared memory address is unused after looking for kAMDGPUSharedMemoryAddrSpace ...let me add that

**Comment by sumesh13** [comment]
Jenkins: retest this please

**Comment by sumesh13** [comment]
Jenkins: retest this please

**Comment by sumesh13** [comment]
jenkins: retest this please

**Comment by whchung** [comment]
@sumesh13 thanks. I've just merged the PR. could you submit it upstream?

**Comment by sumesh13** [comment]
Sure.....will  plan to do it after PR-28270 and math functions.

**Pull request merged**